### ISSUE TITLE
SwaggerParameter not generating data for FromBody parameter

### ISSUE THREAD

**Issue by Thaoden** [domaindrivendev/Swashbuckle.AspNetCore#1579]
I'm currently using `Swashbuckle.AspNetCore` and `Swashbuckle.AspNetCore.Annotations` both in version `5.1.0`.

I have set up a parameter annotation like so: 
``` c#
public async Task<ActionResult<DisplayGroupDto>> SetRuleParts(
    [SwaggerParameter("The import ID of the group to set the rule parts to", Required = true)]
    string groupImportId,
    [FromBody, SwaggerParameter("A list of ruleParts", Required = true)]
    List<CreateRulePartDto> ruleParts)
    {
```

However the description for the `[FromBody]` parameter is not generated:

```
    "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "type": "array",
                "items": {
                  "$ref": "#/components/schemas/CreateRulePartDto"
                },
                "nullable": true
              }
            },
            "text/json": {
              "schema": {
                "type": "array",
                "items": {
                  "$ref": "#/components/schemas/CreateRulePartDto"
                },
                "nullable": true
              }
            },
            "application/*+json": {
              "schema": {
                "type": "array",
                "items": {
                  "$ref": "#/components/schemas/CreateRulePartDto"
                },
                "nullable": true
              }
            }
          }
        },
```

**Comment by domaindrivendev** [comment]
The `SwaggerParameterAttribute` is for enriching the generated `parameters` in the Swagger/OpenAPI document whereas you're wanting to enrich the `requestBody`. It seems the Annotations library should have a `SwaggerRequestBodyAttribute` to fill the gap you've identified. Interested in subnmitting a PR for this? (hint: A custom `IRequestBodyFilter` would make this very simple).

**Comment by Thaoden** [comment]
I thought it was already in it, as it works with eg `FromQuery`. I can have a look on the weekend if I can submit a PR!

**Comment by domaindrivendev** [comment]
`[FromQuery]` is generated as a "parameter" in the generated Swagger/OpenAPI document, hence why the `SwaggerParameterAttribute` works in that case. `[FromBody]` on the other hand is generated as a "requestBody". In other words, the `SwaggerParameterAttribute` is only applicable for _C# parameters_ (note the distinction) that are ultimately mapped to a "parameter" in the generated Swagger/OpenAPI document.

**Comment by domaindrivendev** [comment]
A solution for this is now merged to master and queued up for the `v5.3.0` release (due next week). There's now a `SwaggerRequestBodyAttribute` that can be applied to "body" bound parameters OR properties:

```csharp
public async Task<ActionResult<DisplayGroupDto>> SetRuleParts(
    [SwaggerParameter("The import ID of the group to set the rule parts to", Required = true)]
    string groupImportId,
    [FromBody, SwaggerRequestBodyParameter("A list of ruleParts", Required = true)]
    List<CreateRulePartDto> ruleParts)
{
    ...
}
```

To use this you'll have to have the `Annotations` package installed and explicitly opted-in as described [here](https://github.com/domaindrivendev/Swashbuckle.AspNetCore#swashbuckleaspnetcoreannotations)

**Comment by Thaoden** [comment]
Sweet, thanks!

**Comment by vanillajonathan** [comment]
You might want to avoid relying on `FromBody` and other such parameter binding attributes and instead use the `ApiController` attribute which does binding source parameter inference. It is much easier.

https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-3.1#apicontroller-attribute

**Comment by Thaoden** [comment]
Thanks for the hint but that's not always possible.

**Comment by pauloortins** [comment]
@vanillajonathan is it possible to use Swashbuckle without the FromBody and FromRoute annotations?

**Comment by vanillajonathan** [comment]
@pauloortins Yes, I never use `FromBody` or `FromRoute` in my projects, I just decorate the controller with the `ApiController` attribute.

**Comment by pauloortins** [comment]
Do you have a sample of a controller because when I use it I can`t see the correct contract:

![image](https://user-images.githubusercontent.com/1145643/122621415-d5160400-d06b-11eb-9d31-8eaf436f55c0.png)

**Comment by vanillajonathan** [comment]
No, not at hand but on-the-top-of-my-head I would write it something like this:
```cs
namespace Example.Controllers
{
    [Produces("application/json")]
    [Route("api/[controller]")]
    [ApiController] // Be sure to include this attribute
    public class StepsController : ControllerBase
    {
        [HttpPost]
        public IActionResult Post(StepsModel model)
        {
            return Ok();
        }
    }
}
```

**Comment by iamveritas** [comment]
> A solution for this is now merged to master and queued up for the `v5.3.0` release (due next week). There's now a `SwaggerRequestBodyAttribute` that can be applied to "body" bound parameters OR properties:
> 
> ```cs
> public async Task<ActionResult<DisplayGroupDto>> SetRuleParts(
>     [SwaggerParameter("The import ID of the group to set the rule parts to", Required = true)]
>     string groupImportId,
>     [FromBody, SwaggerRequestBodyParameter("A list of ruleParts", Required = true)]
>     List<CreateRulePartDto> ruleParts)
> {
>     ...
> }
> ```
> 
> To use this you'll have to have the `Annotations` package installed and explicitly opted-in as described [here](https://github.com/domaindrivendev/Swashbuckle.AspNetCore#swashbuckleaspnetcoreannotations)

was this SwaggerRequestBodyParameter attribute retired? I can't find it anymore.
I am facing the same issue, the FromBody attribute is ignored completely and the param is generated as query param.
thoughts?
thank you
